.github/workflows/tests_linux_x86.yml

```diff
@@ -0,0 +1,168 @@
+name: Tests::Linux::x86_64
+on:
+  release:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  COVERAGE_FLAGS: "--cov=pennylane_lightning_gpu --cov-report=term-missing --cov-report=xml:./coverage.xml --no-flaky-report -p no:warnings --tb=native" 
+  GCC_VERSION: 11
+  OMP_NUM_THREADS: "2"
+  CI_CUDA_ARCH: 86
+
+concurrency:
+  group: gpu-test-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
One question here. How does this work if we have more than one PR CI/CD action happening at the same time?
Also considering that this workflow will run in a self-hosted runner.

Reply:
It goes by the group name, so in this case since the ending is `${{ github.ref }}` it is per PR. So if we have a push to a branch while a job is running for the same branch, the existing job will get cancelled and a new job spawned. Actions across PRs aren't as affected. The main concurrency limit we may hit is the max number of runners available to us. You can read more on concurrency groups here: https://docs.github.com/en/actions/using-jobs/using-concurrency

Reply:
Thank you @rashidnhm, sorry if my question was not clear.
I'm aware of how the concurrency groups work, but I'm wondering about our self-hosted runners' resources.
If we ask for more jobs than we have runners for, in a single PR, I understand that some will be pending.
In the case of different PRs, will GH manage that exactly like it does with its own runners?
In this case, will the other PRs have their workloads pending? Or should we expect something different?

Reply:
Ah sorry @AmintorDusko I did misunderstand the question. You are correct in your understanding. The number of runners we have is shared across the entire repo. So if we have different PRs all triggering jobs around the same time, you will see them be pending until a resource frees up to take the job. If we see this behaviour happen to quickly or jobs being pending for far too long, we can increase the maximum number of runners we have.

Reply:
According to @rashidnhm this operates the same as `      - name: Cancel Previous Runs`. I'll remove one for the other.

Reply:
Make sense @mlxd. Thank you for that!